crates/pallet-domains/src/lib.rs

```diff
@@ -574,8 +576,16 @@ mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
-    pub fn successful_bundles() -> Vec<H256> {
-        SuccessfulBundles::<T>::get()
+    pub fn successful_bundles(domain_id: DomainId) -> Vec<H256> {
+        SuccessfulBundles::<T>::get(domain_id)
+    }
+
+    pub fn successful_bundles_of_all_domains() -> Vec<H256> {
```

Comment:
Not for performance benefit but for use case in the bundle validator, which needs to keep track of all the successfully submitted bundles of the last K consensus block.

Reply:
This is giving all the domain's budles. Why would every domain client would need to store these ?

Reply:
The bundle validator exists in the consensus node not domain node.

Reply:
The check in the transaction pool ensures no duplicated bundles before they are included in the block, the runtime check ensures no duplicated bundles from the previous block can be included again. Once a bundle X is included in the block, and then you submit a bundle duplicate to X, which was created against a previous consensus block, it will be definitely rejected as stale in the runtime, no?

Reply:
> the runtime check ensures no duplicated bundles from the previous block can be included again

I'm not sure if we have this check now, do you mean the check in `pre_dispatch/validate_unsigned` or the stale receipt check?